src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
```

Comment:
Will do.

On a side note, calling `InternetQueryOptionA` with a nullptr for buffer, makes it return `false` and that's why there's no check. It does however set the size ptr.

Do you prefer a fresh PR with a single commit?